mail_sendgrid_mass_mailing/tests/test_mass_mailing.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+# © 2017 Emanuel Cino - <ecino@compassion.ch>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import mock
+from odoo.tests.common import TransactionCase
+
+mock_sendgrid_api_client = ('odoo.addons.mail_sendgrid.models.mail_mail'
+                            '.SendGridAPIClient')
+mock_config = ('odoo.addons.mail_sendgrid.models.mail_mail.'
+               'config')
+
+
+class FakeClient(object):
+    """ Mock Sendgrid APIClient """
+    status_code = 202
+    body = 'ok'
+
+    def __init__(self):
+        self.client = self
+        self.mail = self
+        self.send = self
+
+    def post(self, **kwargs):
+        return self
+
+
+class FakeRequest(object):
+    """ Simulate a Sendgrid JSON request """
+    def __init__(self, data):
+        self.jsonrequest = [data]
+
+
+class TestMailSendgrid(TransactionCase):
+    def setUp(self):
```

Comment:
same: try using `setUpClass`

Reply:
I tried with it but I get many errors in tearDown

Reply:
you have to use `common.SavepointCase` as base class instead.

Reply:
Yes I tried that, but then I don't know what the errors are about in the tearDown method.

Reply:
which errors? do you have a traceback?

Reply:
Ok, that was my bad, I forgot to add the `@classmethod` decorators and change tearDown to tearDownClass as well.

I still couldn't change the test of Sendgrid because of the `HttpCase` class.